### ISSUE TITLE
Cannot find arm-none-eabi-gcc

### ISSUE THREAD

**Issue by pbauszat** [NessieCircuits/Riotee_AppTemplate#2]
When compiling with Visual Studio code on Windows, the ARM gcc compiler cannot be found:

`Unable to resolve configuration with compilerPath "/arm-none-eabi-gcc"`

The compiler directory is in _$PATH$_ and can be run from the terminal and powershell.
The problem only occurs when the **gnuInstallRoot** setting is left empty.

It seems that the issue is that VS code tries to run

`/arm-none-eabi-gcc`

instead of just

`arm-none-eabi-gcc`

**Comment by geissdoerfer** [comment]
I can't reproduce this on Windows 10. Everything works fine with the following `settings.json`:
```
{
    "rioteeSdkRoot": "C:/Uers/Myuser/Riotee_SDK",
    "operSys": "Win32",
    "gnuInstallRoot": "",
    "files.associations": {
        "type_traits": "c",
    },
    "editor.formatOnSave": true,
}
```

and this in the Path variable: `C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\12.2 mpacbti-rel1\bin`

 Can you please post your `settings.json`and your PATH variable? Also, please insert `$(info    PREFIX is $(PREFIX))` in the SDK's Makefile right before the line that starts with `.PHONY`and post the output from terminal when trying to build the application.

**Comment by pbauszat** [comment]
Here is the content of my settings.json (I set the Riotee SDK path relative and that works fine):
```
{
    "rioteeSdkRoot": "./../../../../Riotee_SDK",
    "sdkRoot": "./../../../../Riotee_SDK",
    "operSys": "Win32",
    "gnuInstallRoot": "",
    "files.associations": {
        "type_traits": "c",
        "riotee_timing.h": "c",
        "riotee_stella.h": "c"
    },
    "editor.formatOnSave": true,
}

```
And these are the warnings in the output after opening the folder in VS code:
```
[1/25/2024, 11:11:07 AM] Unable to resolve configuration with compilerPath "/arm-none-eabi-gcc".  Using "cl.exe" instead.
[1/25/2024, 11:11:07 AM] Unable to resolve configuration with compilerPath "/arm-none-eabi-gcc".  Using "cl.exe" instead.
```

This is the extra output when adding the line to the SDK's makefile:
`PREFIX is ""arm-none-eabi-`

**Comment by geissdoerfer** [comment]
Ah I see, it's just the linting etc in VSC. Compliation works fine, right? I found the problem, will think of a solution.

**Comment by geissdoerfer** [comment]
I changed a definition which may fix your problem. Unfortunately, I ran into another issue that prevents IntelliSense from finding the compiler at least on on Linux. I've reported the issue here: https://github.com/microsoft/vscode-cpptools/issues/11903

Please check if 7cf076f05d8d71011c9717fe3b7726bb06383904 fixed your problem. If not, please add the path to your compiler under `gnuInstallRoot`in `settings.json` with a trailing slash. That should make IntelliSense work for you.

**Comment by pbauszat** [comment]
The proposed bugfix does not solve the problem, because VS code then looks for the compiler in the current project directory (at least on Windows). 

However, setting `gnuInstallRoot` does fix the problem. It's not a great solution because the configuration file now contains a hard-coded local path (hard to share code between machines) but it works for now and compilation is fine anyways as you mentioned.

Thanks! I'll close the issue for now.

**Comment by geissdoerfer** [comment]
FYI: This bug has been fixed in vscode 1.19 and you should no longer get the error message.
